asserts/database.go

```diff
@@ -27,22 +27,34 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
+	"time"
 
 	"golang.org/x/crypto/openpgp/packet"
 
 	"github.com/ubuntu-core/snappy/helpers"
 )
 
+// PublicKey is a public key as used by the assertion database.
+type PublicKey interface {
+	PublicKeyPrim
```

Comment:
I'm missing context about the underlining design here. Gut feeling is that this should be simply PublicKey, but I'm not quite understanding the goal of these *Prim interfaces. I hope we can do without them.